Manipulators/StandardManipulator.cs

```diff
@@ -52,40 +36,68 @@ void Update()
 			}
 		}
 
-		void OnTranslateDragging(BaseHandle handle, HandleEventData eventData)
+		protected override void OnHandleHoverStarted(BaseHandle handle, HandleEventData eventData)
 		{
-			ConstrainedAxis constraints = 0;
-			var constrainedHandle = handle as IAxisConstraints;
-			if (constrainedHandle != null)
-				constraints = constrainedHandle.constraints;
+			base.OnHandleHoverStarted(handle, eventData);
 
-			translate(eventData.deltaPosition, eventData.rayOrigin, constraints);
+			if (!handle.hasDragSource && handle is RadialHandle)
+				handle.GetComponent<MeshFilter>().sharedMesh = m_FatRadialHandleMesh;
 		}
 
-		void OnRotateDragging(BaseHandle handle, HandleEventData eventData)
+		protected override void OnHandleHoverEnded(BaseHandle handle, HandleEventData eventData)
 		{
-			rotate(eventData.deltaRotation, eventData.rayOrigin);
+			base.OnHandleHoverEnded(handle, eventData);
+
+			if (!handle.hasDragSource && !handle.hasHoverSource && handle is RadialHandle)
+				handle.GetComponent<MeshFilter>().sharedMesh = m_RadialHandleMesh;
 		}
 
-		void OnHandleDragStarted(BaseHandle handle, HandleEventData eventData)
+		protected override void OnHandleDragStarted(BaseHandle handle, HandleEventData eventData)
 		{
-			foreach (var h in m_AllHandles)
-				h.gameObject.SetActive(h == handle);
+			base.OnHandleDragStarted(handle, eventData);
 
-			OnDragStarted();
+			if (handle.IndexOfDragSource(eventData.rayOrigin) > 0)
+				return;
 
-			dragging = true;
+			if (handle is SphereHandle)
+				handle.transform.localScale *= m_SphereHandleHideScale;
 		}
 
-		void OnHandleDragEnded(BaseHandle handle, HandleEventData eventData)
+		protected override void OnHandleDragging(BaseHandle handle, HandleEventData eventData)
 		{
-			if (gameObject.activeSelf)
-				foreach (var h in m_AllHandles)
-					h.gameObject.SetActive(true);
+			base.OnHandleDragging(handle, eventData);
+
+			var rayOrigin = eventData.rayOrigin;
+			if (handle.IndexOfDragSource(rayOrigin) > 0)
+				return;
+
+			if (handle is RadialHandle)
+			{
+				rotate(eventData.deltaRotation, rayOrigin);
+			}
+			else
+			{
+				ConstrainedAxis constraints = 0;
+				var constrainedHandle = handle as IAxisConstraints;
+				if (constrainedHandle != null)
+					constraints = constrainedHandle.constraints;
+
+				translate(eventData.deltaPosition, rayOrigin, constraints);
+			}
+		}
+
+		protected override void OnHandleDragEnded(BaseHandle handle, HandleEventData eventData)
+		{
+			base.OnHandleDragEnded(handle, eventData);
+
+			if (handle.IndexOfDragSource(eventData.rayOrigin) > 0)
+				return;
 
-			OnDragEnded(eventData.rayOrigin);
+			if (handle is SphereHandle)
```

Comment:
I see this as part of the StandardManipulator. The SphereHandle and RadialHandle provide a way of processing drag events along the surface of a sphere, or into rotation deltas. As with the rest of this PR, I tried to keep any visual behavior out of the Handle classes themselves.

It helps to remember that the sphere object that renders the object with the spherehandle on it has nothing to do with the "sphere" in spherehandle. That sphere is the sphere that encompasses the controller and defines the surface along which you will drag the object.